### ISSUE TITLE
[BUG] AutoARIMA outputs flat predictions for close to zero data

### ISSUE THREAD

**Issue by claudia-hm** [alan-turing-institute/sktime#2075]
AutoARIMA is returning constant predictions when data is too small., i.e., close to zero.

Initially, I generated a linear trendy time series with slope 0.5 and intercept 100, plus some noise.
```python
a, b = 0.5, 100
y1 = a*np.arange(0,100)+b
noise = np.random.normal(y1)
y1 = y1 + noise
```
To obtain forecasts with AutoARIMA I used the following code:
```python
from sktime.forecasting.arima import AutoARIMA
from sktime.utils.plotting import plot_series

window = 20

y1_train = pd.Series(y1[:-window], index = pd.Index(np.arange(0, len(y1)-window)))
y1_test = pd.Series(y1[-window:], index = pd.Index(np.arange(len(y1)-window, len(y1))))

fh = np.arange(1, window+1)
model = AutoARIMA()
model.fit(y1_train, fh=fh)
y1_pred = model.predict(fh)
plot_series(y1_train, y1_test, y1_pred, labels=["train", "test", "pred"])
```
which gave me the following prediction:
![image](https://user-images.githubusercontent.com/59530433/154240932-bcf7b324-7ac3-457b-8d3d-15b6cf26362a.png)

The bug appears when I transform the original time series by dividing it by a high value. I needed to do this to change units.
```python
y1 = y1/(10 ** 6)
```
This time, when I runned the code above for computing forecasts with AutoARIMA the result is:
![image](https://user-images.githubusercontent.com/59530433/154241448-b8a8224b-d8d1-4794-8ca2-52aacf1b34f5.png)

and the returned time series is:
```python
80    0.00028
81    0.00028
82    0.00028
83    0.00028
84    0.00028
85    0.00028
86    0.00028
87    0.00028
88    0.00028
89    0.00028
90    0.00028
91    0.00028
92    0.00028
93    0.00028
94    0.00028
95    0.00028
96    0.00028
97    0.00028
98    0.00028
99    0.00028
dtype: float64
```
I expected to obtain a similar forecast to that one of the first image, as the time series is nearly the same. The only transformation that I applied is a change of units.

**Versions**
These are the dependencies and versions:

```bash
(debug_sktime) claudia@claudia-hm:~/Desktop/debug_sktime$ python --version
Python 3.8.10
(debug_sktime) claudia@claudia-hm:~/Desktop/debug_sktime$ pip freeze
cycler==0.11.0
Cython==0.29.27
Deprecated==1.2.13
fonttools==4.29.1
joblib==1.1.0
kiwisolver==1.3.2
llvmlite==0.38.0
matplotlib==3.5.1
numba==0.55.1
numpy==1.21.5
packaging==21.3
pandas==1.4.1
patsy==0.5.2
Pillow==9.0.1
pmdarima==1.8.4
pyparsing==3.0.7
python-dateutil==2.8.2
pytz==2021.3
scikit-learn==1.0.2
scipy==1.8.0
seaborn==0.11.2
six==1.16.0
sktime==0.10.0
statsmodels==0.13.2
threadpoolctl==3.1.0
urllib3==1.26.8
wrapt==1.13.3

```

**Comment by fkiraly** [comment]
@claudia-hm, this might be a "numerical accuracy" issue with `AutoARIMA` from `pmdarima`, which `sktime` directly interfaces (with a "nicer" since uniform interface).

My suspicion would be that there is some arithmetic happening where small numbers are getting added to large numbers, then the same large number is subtracted, or similar, so the small number becomes numerically zero.

Have you tried whether direct application of `pmdarima` `AutoARIMA` gives you the same behaviour? If so, that would speak for this hypothesis.

**Comment by claudia-hm** [comment]
Hello @fkiraly,

I've just tried an equivalent code with ```pmdarima``` and yes, I obtain the same behavior.
```python
import pmdarima as pm
from sktime.utils.plotting import plot_series
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 

#generate data
a, b = 0.5, 100
y1 = a*np.arange(0,100)+b
noise = np.random.normal(y1)
y1 = y1 + noise
y1 = y1 / (10**6)

window = 20
y1_train = pd.Series(y1[:-window], index = pd.Index(np.arange(0, len(y1)-window)))
y1_test = pd.Series(y1[-window:], index = pd.Index(np.arange(len(y1)-window, len(y1))))

# Fit your model
model = pm.auto_arima(y1_train)

# make your forecasts
forecasts = model.predict(y1_test.shape[0])  # predict N steps into the future
y1_pred = pd.Series(forecasts, index=y1_test.index)
plot_series(y1_train, y1_test, y1_pred, labels=["train", "test", "pred"])
plt.show()
```
![image](https://user-images.githubusercontent.com/59530433/154475232-47553196-74f5-490b-b669-ffb6927defa5.png)

I coincide that it has to be some kind of numerical issue. However, maybe there should be some kind of tolerance value that could prevent this behavior. I'll open an issue in the ```pmdarima``` repo.

Thanks!

**Comment by aiwalter** [comment]
@claudia-hm pls link issue here

**Comment by fkiraly** [comment]
>I coincide that it has to be some kind of numerical issue. However, maybe there should be some kind of tolerance value that could prevent this behavior

Probably. Although, the purpose of the class `AutoARIMA` is to directly interface `pmdarima`.
So whatever `pmdarima` outputs, whether it's very sensible or garbage, will be returned.

Based on your finding that it's the same in `pmdarima`, I'd assume it's an oversight in `pmdarima`, so the best way would be to have it fixed there. Issue link appreciated.

**Comment by claudia-hm** [comment]
Great. 

You can find the ```pmdarima``` issue here: https://github.com/alkaline-ml/pmdarima/issues/480

**Comment by fkiraly** [comment]
Excellent! let's see what they say, I'm curious...

**Comment by fkiraly** [comment]
just following up on this, @claudia-hm. No reaction from `pmdarima` on the issue yet? E.g., confirmation whether it´s a bug?
Have you heard from them otherwise?

**Comment by fkiraly** [comment]
Hm, they have not answered ny other issue either - https://github.com/alkaline-ml/pmdarima/issues/492
Is `pmdarima` still alive??

**Comment by benHeid** [comment]
This bug still exist. Code of @claudia-hm is still reproducing the bug. Or the following Claude Code generated Code without Seaborn dependency. 

<details>

```python

import numpy as np
import pandas as pd
from sktime.forecasting.arima import AutoARIMA

np.random.seed(42)

# Reproduce the original repro's data generation (kept intentionally as in the
# issue: noise is drawn with the trend values used as the normal distribution's
# mean, per the reporter's original code), just seeded for determinism.
a, b = 0.5, 100
n = 100
t = np.arange(0, n)
y1 = a * t + b
noise = np.random.normal(y1)
y1 = y1 + noise

# Scale down by 1e6, as in the bug report -- this is what triggered the flat
# forecast bug in pmdarima/AutoARIMA.
y1_scaled = y1 / (10 ** 6)

window = 20
y_train = pd.Series(
    y1_scaled[:-window], index=pd.Index(np.arange(0, len(y1_scaled) - window))
)

fh = np.arange(1, window + 1)

model = AutoARIMA(suppress_warnings=True)
model.fit(y_train, fh=fh)
y_pred = model.predict(fh)

pred_values = y_pred.to_numpy()
pred_std = np.std(pred_values)
pred_range = pred_values.max() - pred_values.min()

print("Predicted values (scaled series):")
print(y_pred)
print(f"std of predictions: {pred_std}")
print(f"range of predictions: {pred_range}")

# The reported bug: predictions collapse to a numerically-constant flat line
# (e.g. all equal to 0.00028) instead of tracking the underlying trend.
# A healthy forecast on a trending series should show real variation across
# the forecast horizon.
assert pred_std > 1e-8, (
    "AutoARIMA produced a flat/constant forecast (std="
    f"{pred_std}) for the scaled-down trending series -- "
    "reproducing the reported numerical-accuracy bug."
)

```

</details>

Triaged with Claude Code
